src/checkers/inference/solver/util/PrintUtils.java

```diff
@@ -40,61 +43,80 @@
+        try {
+            return new PrintStream(new FileOutputStream(file, appendWrite));
+        } catch (FileNotFoundException e) {
+            throw new BugInCF("Cannot find file " + file);
         }
-        System.out.println(printBuffer.toString());
-        System.out.flush();
-        System.out.println("/**********************************************************/");
     }
 
-    public static void writeResult(Map<Integer, AnnotationMirror> result) {
+    /**
+     * Creates and returns a string representation of inference solutions, where each row shows the id and the annotation of a slot.
+     * @param solutions inference solutions: a map between slot IDs and annotation mirrors
+     * @return string representation of inference solutions
+     */
+    private static String generateSolutionsString(Map<Integer, AnnotationMirror> solutions) {
+        final AnnotatedTypeFactory atf = InferenceMain.getInstance().getRealTypeFactory();
 
+        // string length of the highest slot ID, used to pad spaces for pretty formatting
         final int maxLength = String.valueOf(InferenceMain.getInstance().getSlotManager().getNumberOfSlots()).length();
-        StringBuilder printResult = new StringBuilder();
 
-        for (Integer j : result.keySet()) {
-            printResult.append("SlotID: ");
-            printResult.append(String.valueOf(j));
-            for (int i = 0; i < maxLength + 2 - String.valueOf(j).length(); i++) {
-                printResult.append(" ");
+        StringBuilder sb = new StringBuilder();
+
+        for (Integer j : solutions.keySet()) {
+            sb.append("SlotID: ");
+            sb.append(j);
+            for (int i = 0; i < maxLength + 2 - j.toString().length(); i++) {
+                sb.append(" ");
             }
-            printResult.append("Annotation: ");
-            printResult.append(result.get(j).toString());
-            printResult.append("\n");
+            sb.append("Annotation: ");
+            sb.append(atf.getAnnotationFormatter().formatAnnotationMirror(solutions.get(j)));
+            sb.append("\n");
         }
 
-        File basePath = new File(new File("").getAbsolutePath());
-        String writePath = basePath.getAbsolutePath() + "/result" + ".txt";
-        File file = new File(writePath);
-        PrintWriter pw;
-        try {
-            pw = new PrintWriter(file);
-            pw.write(printResult.toString());
-            pw.close();
-        } catch (FileNotFoundException e) {
-            e.printStackTrace();
-        }
+        return sb.toString();
+    }
 
-        System.out.println("Result has been written to: " + writePath);
+    /**
+     * Outputs the solutions to the given stream.
+     * @param stream
+     * @param solutions
+     */
+    private static void outputSolutions(PrintStream stream, Map<Integer, AnnotationMirror> solutions) {
+        stream.println("/***********************Solutions**************************/");
```

Comment:
move /**....*/ lines into gen solution string
(merge gen solution string with this method)